src/main/java/hudson/plugins/git/extensions/impl/SubmoduleOption.java

```diff
@@ -42,15 +42,17 @@
     private String reference;
     private boolean parentCredentials;
     private Integer timeout;
+    private Integer threads;
 
     @DataBoundConstructor
-    public SubmoduleOption(boolean disableSubmodules, boolean recursiveSubmodules, boolean trackingSubmodules, String reference,Integer timeout, boolean parentCredentials) {
+    public SubmoduleOption(boolean disableSubmodules, boolean recursiveSubmodules, boolean trackingSubmodules, String reference, Integer timeout, boolean parentCredentials, Integer threads) {
```

Comment:
The constructor should only be extended for mandatory parameters.
For this optional parameter a setter with `@DataBoundSetter` should be used.

Reply:
@darxriggs Ok, I've made this change and amended it to the previous commit. The tests still pass, so I hope that I did it right. 😅

Reply:
I tend to update primary constructors like the following:

```java
@DataBoundConstructor
public SubmoduleOption(boolean disableSubmodules, boolean recursiveSubmodules, boolean trackingSubmodules, String reference, Integer timeout, boolean parentCredentials, Integer threads) {
    // cool code goes here
}

//but we still need to migrate older configurations so...
@Deprecated
public SubmoduleOption(boolean disableSubmodules, boolean recursiveSubmodules, boolean trackingSubmodules, String reference,Integer timeout, boolean parentCredentials) {
    this(disableSubmodules, recursiveSubmodules, trackingSubmodules, reference, timeout, parentCredentials, 1);
    // don't forget the 1 is to set migrating configurations to 1 thread by default.
}
```